src/instrumentsscene/view/parttreeitem.cpp

```diff
@@ -48,7 +48,7 @@ void PartTreeItem::init(const notation::Part* masterPart)
     }
 
     setId(part->id());
-    setTitle(part->partName().isEmpty() ? part->instrument()->name() : part->partName());
+    setTitle(part->instrument()->name().isEmpty() ? part->partName() : part->instrument()->name());
```

Comment:
I believe the intrument name can never be empty, only the part name can, at least that's how it was in 3.x IIRC